pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, please use the tmp_path fixture or tmp_path_factory.mktemp

<!--
Welcome to the docker-compose issue tracker! Before creating an issue, please heed the following:

1. This tracker should only be used to report bugs and request features / enhancements to docker-compose
    - For questions and general support, use https://forums.docker.com
    - For documentation issues, use https://github.com/docker/docker.github.io
    - For issues with the `docker stack` commands and the version 3 of the Compose file, use
      https://github.com/docker/cli
2. Use the search function before creating a new issue. Duplicates will be closed and directed to
   the original discussion.
3. When making a bug report, make sure you provide all required information. The easier it is for
   maintainers to reproduce, the faster it'll be fixed.
-->

## Description of the issue
Several unit tests fail with an error regarding deprecated and removed functionality in pytest 4. It is not currently possible to run some tests with pytest 4 or later.

## Context information (for bug reports)

**Output of `docker-compose version`**
```
docker-compose version 1.24.1, build 4667896
```

**Output of `docker version`**
Not applicable

**Output of `docker-compose config`**
(Make sure to add the relevant `-f` and other flags)
Not applicable

## Steps to reproduce the issue

1. Create new virtualenv with python 3.7, `pip install -r requirements.txt`
2. Run `pip install pytest`
3. Run `py.test --verbose tests/unit`

### Observed result
```
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
```

### Expected result
These tests should pass.

### Stacktrace / full error message
```
=================================== FAILURES ===================================
________________ ConfigTest.test_load_mixed_extends_resolution _________________
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
____ ConfigTest.test_load_with_multiple_files_and_extends_in_override_file _____
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
______________________ ConfigTest.test_load_yaml_with_bom ______________________
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
__________________ ConfigTest.test_load_yaml_with_yaml_error ___________________
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
_____________ ExtendsTest.test_extends_with_defined_version_passes _____________
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
___________________ ExtendsTest.test_extends_with_depends_on ___________________
tests/unit/config/config_test.py:4774: in test_extends_with_depends_on
    tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
___________ ExtendsTest.test_extends_with_environment_and_env_files ____________
tests/unit/config/config_test.py:4664: in test_extends_with_environment_and_env_files
    tmpdir = py.test.ensuretemp('test_extends_with_environment')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4774: in test_extends_with_depends_on
    tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
____________ ExtendsTest.test_extends_with_mixed_versions_is_error _____________
tests/unit/config/config_test.py:4728: in test_extends_with_mixed_versions_is_error
    tmpdir = py.test.ensuretemp('test_extends_with_mixed_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
/usr/lib64/python3.7/unittest/mock.py:1621: in _inner
    return f(*args, **kw)
tests/unit/config/config_test.py:4664: in test_extends_with_environment_and_env_files
    tmpdir = py.test.ensuretemp('test_extends_with_environment')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4774: in test_extends_with_depends_on
    tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
_____________________ ExtendsTest.test_extends_with_ports ______________________
tests/unit/config/config_test.py:4807: in test_extends_with_ports
    tmpdir = py.test.ensuretemp('test_extends_with_ports')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4728: in test_extends_with_mixed_versions_is_error
    tmpdir = py.test.ensuretemp('test_extends_with_mixed_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
/usr/lib64/python3.7/unittest/mock.py:1621: in _inner
    return f(*args, **kw)
tests/unit/config/config_test.py:4664: in test_extends_with_environment_and_env_files
    tmpdir = py.test.ensuretemp('test_extends_with_environment')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4774: in test_extends_with_depends_on
    tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
__________________ ExtendsTest.test_extends_with_security_opt __________________
tests/unit/config/config_test.py:4829: in test_extends_with_security_opt
    tmpdir = py.test.ensuretemp('test_extends_with_ports')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4807: in test_extends_with_ports
    tmpdir = py.test.ensuretemp('test_extends_with_ports')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4728: in test_extends_with_mixed_versions_is_error
    tmpdir = py.test.ensuretemp('test_extends_with_mixed_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
/usr/lib64/python3.7/unittest/mock.py:1621: in _inner
    return f(*args, **kw)
tests/unit/config/config_test.py:4664: in test_extends_with_environment_and_env_files
    tmpdir = py.test.ensuretemp('test_extends_with_environment')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4774: in test_extends_with_depends_on
    tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
_________________ EnvironmentTest.test_env_vars_from_file_bom __________________
tests/unit/config/environment_test.py:49: in test_env_vars_from_file_bom
    tmpdir = pytest.ensuretemp('env_file')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4829: in test_extends_with_security_opt
    tmpdir = py.test.ensuretemp('test_extends_with_ports')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4807: in test_extends_with_ports
    tmpdir = py.test.ensuretemp('test_extends_with_ports')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4728: in test_extends_with_mixed_versions_is_error
    tmpdir = py.test.ensuretemp('test_extends_with_mixed_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
/usr/lib64/python3.7/unittest/mock.py:1621: in _inner
    return f(*args, **kw)
tests/unit/config/config_test.py:4664: in test_extends_with_environment_and_env_files
    tmpdir = py.test.ensuretemp('test_extends_with_environment')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4774: in test_extends_with_depends_on
    tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
______________ EnvironmentTest.test_env_vars_from_file_whitespace ______________
tests/unit/config/environment_test.py:58: in test_env_vars_from_file_whitespace
    tmpdir = pytest.ensuretemp('env_file')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/environment_test.py:49: in test_env_vars_from_file_bom
    tmpdir = pytest.ensuretemp('env_file')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4829: in test_extends_with_security_opt
    tmpdir = py.test.ensuretemp('test_extends_with_ports')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4807: in test_extends_with_ports
    tmpdir = py.test.ensuretemp('test_extends_with_ports')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4728: in test_extends_with_mixed_versions_is_error
    tmpdir = py.test.ensuretemp('test_extends_with_mixed_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
/usr/lib64/python3.7/unittest/mock.py:1621: in _inner
    return f(*args, **kw)
tests/unit/config/config_test.py:4664: in test_extends_with_environment_and_env_files
    tmpdir = py.test.ensuretemp('test_extends_with_environment')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4774: in test_extends_with_depends_on
    tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:4750: in test_extends_with_defined_version_passes
    tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1744: in test_load_yaml_with_yaml_error
    tmpdir = py.test.ensuretemp('invalid_yaml_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:1757: in test_load_yaml_with_bom
    tmpdir = py.test.ensuretemp('bom_yaml')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:759: in test_load_with_multiple_files_and_extends_in_override_file
    tmpdir = py.test.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
tests/unit/config/config_test.py:794: in test_load_mixed_extends_resolution
    tmpdir = pytest.ensuretemp('config_test')
venv/lib/python3.7/site-packages/_pytest/tmpdir.py:98: in ensuretemp
    warnings.warn(PYTEST_ENSURETEMP, stacklevel=2)
E   pytest.RemovedInPytest4Warning: pytest/tmpdir_factory.ensuretemp is deprecated, 
E   please use the tmp_path fixture or tmp_path_factory.mktemp
```

## Additional information

Tested on Fedora 30, with docker-compose built from source tarball. This issue surfaced when attempting to package the latest version for Fedora 31 which uses a later version of pytest than is listed in requirements-dev.txt.

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -2,5 +2,6 @@ coverage==4.5.4
 ddt==1.2.2
 flake8==3.7.9
 mock==3.0.5
-pytest==3.6.3
+pytest==5.1.1; python_version >= '3.5'
+pytest==4.6.5; python_version < '3.5'
 pytest-cov==2.8.1

--- a/tests/helpers.py
+++ b/tests/helpers.py
@@ -1,6 +1,7 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+import contextlib
 import os
 
 from compose.config.config import ConfigDetails
@@ -55,3 +56,17 @@ def create_host_file(client, filename):
         content = fh.read()
 
     return create_custom_host_file(client, filename, content)
+
+
+@contextlib.contextmanager
+def cd(path):
+    """
+    A context manager which changes the working directory to the given
+    path, and then changes it back to its previous value on exit.
+    """
+    prev_cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(prev_cwd)

--- a/tests/integration/project_test.py
+++ b/tests/integration/project_test.py
@@ -8,7 +8,6 @@ import random
 import shutil
 import tempfile
 
-import py
 import pytest
 from docker.errors import APIError
 from docker.errors import NotFound
@@ -16,6 +15,7 @@ from docker.errors import NotFound
 from .. import mock
 from ..helpers import build_config as load_config
 from ..helpers import BUSYBOX_IMAGE_WITH_TAG
+from ..helpers import cd
 from ..helpers import create_host_file
 from .testcases import DockerClientTestCase
 from .testcases import SWARM_SKIP_CONTAINERS_ALL
@@ -1329,9 +1329,9 @@ class ProjectTest(DockerClientTestCase):
             })
         details = config.ConfigDetails('.', [base_file, override_file])
 
-        tmpdir = py.test.ensuretemp('logging_test')
-        self.addCleanup(tmpdir.remove)
-        with tmpdir.as_cwd():
+        tmpdir = tempfile.mkdtemp('logging_test')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with cd(tmpdir):
             config_data = config.load(details)
         project = Project.from_config(
             name='composetest', config_data=config_data, client=self.client

--- a/tests/integration/state_test.py
+++ b/tests/integration/state_test.py
@@ -6,8 +6,10 @@ from __future__ import absolute_import
 from __future__ import unicode_literals
 
 import copy
+import os
+import shutil
+import tempfile
 
-import py
 from docker.errors import ImageNotFound
 
 from ..helpers import BUSYBOX_IMAGE_WITH_TAG
@@ -426,29 +428,32 @@ class ServiceStateTest(DockerClientTestCase):
 
     @no_cluster('Can not guarantee the build will be run on the same node the service is deployed')
     def test_trigger_recreate_with_build(self):
-        context = py.test.ensuretemp('test_trigger_recreate_with_build')
-        self.addCleanup(context.remove)
+        context = tempfile.mkdtemp('test_trigger_recreate_with_build')
+        self.addCleanup(shutil.rmtree, context)
 
         base_image = "FROM busybox\nLABEL com.docker.compose.test_image=true\n"
-        dockerfile = context.join('Dockerfile')
-        dockerfile.write(base_image)
+        dockerfile = os.path.join(context, 'Dockerfile')
+        with open(dockerfile, mode="w") as dockerfile_fh:
+            dockerfile_fh.write(base_image)
 
         web = self.create_service('web', build={'context': str(context)})
         container = web.create_container()
 
-        dockerfile.write(base_image + 'CMD echo hello world\n')
+        with open(dockerfile, mode="w") as dockerfile_fh:
+            dockerfile_fh.write(base_image + 'CMD echo hello world\n')
         web.build()
 
         web = self.create_service('web', build={'context': str(context)})
         assert ('recreate', [container]) == web.convergence_plan()
 
     def test_image_changed_to_build(self):
-        context = py.test.ensuretemp('test_image_changed_to_build')
-        self.addCleanup(context.remove)
-        context.join('Dockerfile').write("""
-            FROM busybox
-            LABEL com.docker.compose.test_image=true
-        """)
+        context = tempfile.mkdtemp('test_image_changed_to_build')
+        self.addCleanup(shutil.rmtree, context)
+        with open(os.path.join(context, 'Dockerfile'), mode="w") as dockerfile:
+            dockerfile.write("""
+                FROM busybox
+                LABEL com.docker.compose.test_image=true
+            """)
 
         web = self.create_service('web', image='busybox')
         container = web.create_container()

--- a/tests/unit/config/config_test.py
+++ b/tests/unit/config/config_test.py
@@ -10,7 +10,6 @@ import tempfile
 from operator import itemgetter
 from random import shuffle
 
-import py
 import pytest
 import yaml
 from ddt import data
@@ -18,6 +17,7 @@ from ddt import ddt
 
 from ...helpers import build_config_details
 from ...helpers import BUSYBOX_IMAGE_WITH_TAG
+from ...helpers import cd
 from compose.config import config
 from compose.config import types
 from compose.config.config import ConfigFile
@@ -780,13 +780,14 @@ class ConfigTest(unittest.TestCase):
             })
         details = config.ConfigDetails('.', [base_file, override_file])
 
-        tmpdir = py.test.ensuretemp('config_test')
-        self.addCleanup(tmpdir.remove)
-        tmpdir.join('common.yml').write("""
-            base:
-              labels: ['label=one']
-        """)
-        with tmpdir.as_cwd():
+        tmpdir = tempfile.mkdtemp('config_test')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with open(os.path.join(tmpdir, 'common.yml'), mode="w") as common_fh:
+            common_fh.write("""
+                base:
+                  labels: ['label=one']
+            """)
+        with cd(tmpdir):
             service_dicts = config.load(details).services
 
         expected = [
@@ -815,19 +816,20 @@ class ConfigTest(unittest.TestCase):
             }
         )
 
-        tmpdir = pytest.ensuretemp('config_test')
-        self.addCleanup(tmpdir.remove)
-        tmpdir.join('base.yml').write("""
-            version: '2.2'
-            services:
-              base:
-                image: base
-              web:
-                extends: base
-        """)
+        tmpdir = tempfile.mkdtemp('config_test')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with open(os.path.join(tmpdir, 'base.yml'), mode="w") as base_fh:
+            base_fh.write("""
+                version: '2.2'
+                services:
+                  base:
+                    image: base
+                  web:
+                    extends: base
+            """)
 
         details = config.ConfigDetails('.', [main_file])
-        with tmpdir.as_cwd():
+        with cd(tmpdir):
             service_dicts = config.load(details).services
             assert service_dicts[0] == {
                 'name': 'prodweb',
@@ -1765,22 +1767,23 @@ class ConfigTest(unittest.TestCase):
         assert services[0]['environment']['SPRING_JPA_HIBERNATE_DDL-AUTO'] == 'none'
 
     def test_load_yaml_with_yaml_error(self):
-        tmpdir = py.test.ensuretemp('invalid_yaml_test')
-        self.addCleanup(tmpdir.remove)
-        invalid_yaml_file = tmpdir.join('docker-compose.yml')
-        invalid_yaml_file.write("""
-            web:
-              this is bogus: ok: what
-        """)
+        tmpdir = tempfile.mkdtemp('invalid_yaml_test')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        invalid_yaml_file = os.path.join(tmpdir, 'docker-compose.yml')
+        with open(invalid_yaml_file, mode="w") as invalid_yaml_file_fh:
+            invalid_yaml_file_fh.write("""
+                web:
+                  this is bogus: ok: what
+            """)
         with pytest.raises(ConfigurationError) as exc:
             config.load_yaml(str(invalid_yaml_file))
 
-        assert 'line 3, column 32' in exc.exconly()
+        assert 'line 3, column 36' in exc.exconly()
 
     def test_load_yaml_with_bom(self):
-        tmpdir = py.test.ensuretemp('bom_yaml')
-        self.addCleanup(tmpdir.remove)
-        bom_yaml = tmpdir.join('docker-compose.yml')
+        tmpdir = tempfile.mkdtemp('bom_yaml')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        bom_yaml = os.path.join(tmpdir, 'docker-compose.yml')
         with codecs.open(str(bom_yaml), 'w', encoding='utf-8') as f:
             f.write('''\ufeff
                 version: '2.3'
@@ -4724,43 +4727,48 @@ class ExtendsTest(unittest.TestCase):
 
     @mock.patch.dict(os.environ)
     def test_extends_with_environment_and_env_files(self):
-        tmpdir = py.test.ensuretemp('test_extends_with_environment')
-        self.addCleanup(tmpdir.remove)
-        commondir = tmpdir.mkdir('common')
-        commondir.join('base.yml').write("""
-            app:
-                image: 'example/app'
-                env_file:
-                    - 'envs'
-                environment:
-                    - SECRET
-                    - TEST_ONE=common
-                    - TEST_TWO=common
-        """)
-        tmpdir.join('docker-compose.yml').write("""
-            ext:
-                extends:
-                    file: common/base.yml
-                    service: app
-                env_file:
-                    - 'envs'
-                environment:
-                    - THING
-                    - TEST_ONE=top
-        """)
-        commondir.join('envs').write("""
-            COMMON_ENV_FILE
-            TEST_ONE=common-env-file
-            TEST_TWO=common-env-file
-            TEST_THREE=common-env-file
-            TEST_FOUR=common-env-file
-        """)
-        tmpdir.join('envs').write("""
-            TOP_ENV_FILE
-            TEST_ONE=top-env-file
-            TEST_TWO=top-env-file
-            TEST_THREE=top-env-file
-        """)
+        tmpdir = tempfile.mkdtemp('test_extends_with_environment')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        commondir = os.path.join(tmpdir, 'common')
+        os.mkdir(commondir)
+        with open(os.path.join(commondir, 'base.yml'), mode="w") as base_fh:
+            base_fh.write("""
+                app:
+                    image: 'example/app'
+                    env_file:
+                        - 'envs'
+                    environment:
+                        - SECRET
+                        - TEST_ONE=common
+                        - TEST_TWO=common
+            """)
+        with open(os.path.join(tmpdir, 'docker-compose.yml'), mode="w") as docker_compose_fh:
+            docker_compose_fh.write("""
+                ext:
+                    extends:
+                        file: common/base.yml
+                        service: app
+                    env_file:
+                        - 'envs'
+                    environment:
+                        - THING
+                        - TEST_ONE=top
+            """)
+        with open(os.path.join(commondir, 'envs'), mode="w") as envs_fh:
+            envs_fh.write("""
+                COMMON_ENV_FILE
+                TEST_ONE=common-env-file
+                TEST_TWO=common-env-file
+                TEST_THREE=common-env-file
+                TEST_FOUR=common-env-file
+            """)
+        with open(os.path.join(tmpdir, 'envs'), mode="w") as envs_fh:
+            envs_fh.write("""
+                TOP_ENV_FILE
+                TEST_ONE=top-env-file
+                TEST_TWO=top-env-file
+                TEST_THREE=top-env-file
+            """)
 
         expected = [
             {
@@ -4783,72 +4791,77 @@ class ExtendsTest(unittest.TestCase):
         os.environ['THING'] = 'thing'
         os.environ['COMMON_ENV_FILE'] = 'secret'
         os.environ['TOP_ENV_FILE'] = 'secret'
-        config = load_from_filename(str(tmpdir.join('docker-compose.yml')))
+        config = load_from_filename(str(os.path.join(tmpdir, 'docker-compose.yml')))
 
         assert config == expected
 
     def test_extends_with_mixed_versions_is_error(self):
-        tmpdir = py.test.ensuretemp('test_extends_with_mixed_version')
-        self.addCleanup(tmpdir.remove)
-        tmpdir.join('docker-compose.yml').write("""
-            version: "2"
-            services:
-              web:
-                extends:
-                  file: base.yml
-                  service: base
-                image: busybox
-        """)
-        tmpdir.join('base.yml').write("""
-            base:
-              volumes: ['/foo']
-              ports: ['3000:3000']
-        """)
-
-        with pytest.raises(ConfigurationError) as exc:
-            load_from_filename(str(tmpdir.join('docker-compose.yml')))
-        assert 'Version mismatch' in exc.exconly()
-
-    def test_extends_with_defined_version_passes(self):
-        tmpdir = py.test.ensuretemp('test_extends_with_defined_version')
-        self.addCleanup(tmpdir.remove)
-        tmpdir.join('docker-compose.yml').write("""
-            version: "2"
-            services:
-              web:
-                extends:
-                  file: base.yml
-                  service: base
-                image: busybox
-        """)
-        tmpdir.join('base.yml').write("""
-            version: "2"
-            services:
+        tmpdir = tempfile.mkdtemp('test_extends_with_mixed_version')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with open(os.path.join(tmpdir, 'docker-compose.yml'), mode="w") as docker_compose_fh:
+            docker_compose_fh.write("""
+                version: "2"
+                services:
+                  web:
+                    extends:
+                      file: base.yml
+                      service: base
+                    image: busybox
+            """)
+        with open(os.path.join(tmpdir, 'base.yml'), mode="w") as base_fh:
+            base_fh.write("""
                 base:
                   volumes: ['/foo']
                   ports: ['3000:3000']
-                  command: top
-        """)
+            """)
 
-        service = load_from_filename(str(tmpdir.join('docker-compose.yml')))
+        with pytest.raises(ConfigurationError) as exc:
+            load_from_filename(str(os.path.join(tmpdir, 'docker-compose.yml')))
+        assert 'Version mismatch' in exc.exconly()
+
+    def test_extends_with_defined_version_passes(self):
+        tmpdir = tempfile.mkdtemp('test_extends_with_defined_version')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with open(os.path.join(tmpdir, 'docker-compose.yml'), mode="w") as docker_compose_fh:
+            docker_compose_fh.write("""
+                version: "2"
+                services:
+                  web:
+                    extends:
+                      file: base.yml
+                      service: base
+                    image: busybox
+            """)
+        with open(os.path.join(tmpdir, 'base.yml'), mode="w") as base_fh:
+            base_fh.write("""
+                version: "2"
+                services:
+                  base:
+                    volumes: ['/foo']
+                    ports: ['3000:3000']
+                    command: top
+            """)
+
+        service = load_from_filename(str(os.path.join(tmpdir, 'docker-compose.yml')))
         assert service[0]['command'] == "top"
 
     def test_extends_with_depends_on(self):
-        tmpdir = py.test.ensuretemp('test_extends_with_depends_on')
-        self.addCleanup(tmpdir.remove)
-        tmpdir.join('docker-compose.yml').write("""
-            version: "2"
-            services:
-              base:
-                image: example
-              web:
-                extends: base
-                image: busybox
-                depends_on: ['other']
-              other:
-                image: example
-        """)
-        services = load_from_filename(str(tmpdir.join('docker-compose.yml')))
+        tmpdir = tempfile.mkdtemp('test_extends_with_depends_on')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with open(os.path.join(tmpdir, 'docker-compose.yml'), mode="w") as docker_compose_fh:
+            docker_compose_fh.write("""
+                version: "2"
+                services:
+                  base:
+                    image: example
+                  web:
+                    extends: base
+                    image: busybox
+                    depends_on: ['other']
+                  other:
+                    image: example
+            """)
+        services = load_from_filename(str(os.path.join(tmpdir, 'docker-compose.yml')))
         assert service_sort(services)[2]['depends_on'] == {
             'other': {'condition': 'service_started'}
         }
@@ -4867,45 +4880,47 @@ class ExtendsTest(unittest.TestCase):
         }]
 
     def test_extends_with_ports(self):
-        tmpdir = py.test.ensuretemp('test_extends_with_ports')
-        self.addCleanup(tmpdir.remove)
-        tmpdir.join('docker-compose.yml').write("""
-            version: '2'
+        tmpdir = tempfile.mkdtemp('test_extends_with_ports')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with open(os.path.join(tmpdir, 'docker-compose.yml'), mode="w") as docker_compose_fh:
+            docker_compose_fh.write("""
+                version: '2'
 
-            services:
-              a:
-                image: nginx
-                ports:
-                  - 80
+                services:
+                  a:
+                    image: nginx
+                    ports:
+                      - 80
 
-              b:
-                extends:
-                  service: a
-        """)
-        services = load_from_filename(str(tmpdir.join('docker-compose.yml')))
+                  b:
+                    extends:
+                      service: a
+            """)
+        services = load_from_filename(str(os.path.join(tmpdir, 'docker-compose.yml')))
 
         assert len(services) == 2
         for svc in services:
             assert svc['ports'] == [types.ServicePort('80', None, None, None, None)]
 
     def test_extends_with_security_opt(self):
-        tmpdir = py.test.ensuretemp('test_extends_with_ports')
-        self.addCleanup(tmpdir.remove)
-        tmpdir.join('docker-compose.yml').write("""
-            version: '2'
+        tmpdir = tempfile.mkdtemp('test_extends_with_ports')
+        self.addCleanup(shutil.rmtree, tmpdir)
+        with open(os.path.join(tmpdir, 'docker-compose.yml'), mode="w") as docker_compose_fh:
+            docker_compose_fh.write("""
+                version: '2'
 
-            services:
-              a:
-                image: nginx
-                security_opt:
-                  - apparmor:unconfined
-                  - seccomp:unconfined
+                services:
+                  a:
+                    image: nginx
+                    security_opt:
+                    - apparmor:unconfined
+                    - seccomp:unconfined
 
-              b:
-                extends:
-                  service: a
-        """)
-        services = load_from_filename(str(tmpdir.join('docker-compose.yml')))
+                  b:
+                    extends:
+                      service: a
+            """)
+        services = load_from_filename(str(os.path.join(tmpdir, 'docker-compose.yml')))
         assert len(services) == 2
         for svc in services:
             assert types.SecurityOpt.parse('apparmor:unconfined') in svc['security_opt']

--- a/tests/unit/config/config_test.py
+++ b/tests/unit/config/config_test.py
@@ -1772,13 +1772,13 @@ class ConfigTest(unittest.TestCase):
         invalid_yaml_file = os.path.join(tmpdir, 'docker-compose.yml')
         with open(invalid_yaml_file, mode="w") as invalid_yaml_file_fh:
             invalid_yaml_file_fh.write("""
-                web:
-                  this is bogus: ok: what
+web:
+    this is bogus: ok: what
             """)
         with pytest.raises(ConfigurationError) as exc:
             config.load_yaml(str(invalid_yaml_file))
 
-        assert 'line 3, column 36' in exc.exconly()
+        assert 'line 3, column 22' in exc.exconly()
 
     def test_load_yaml_with_bom(self):
         tmpdir = tempfile.mkdtemp('bom_yaml')

--- a/tests/unit/config/environment_test.py
+++ b/tests/unit/config/environment_test.py
@@ -4,6 +4,9 @@ from __future__ import print_function
 from __future__ import unicode_literals
 
 import codecs
+import os
+import shutil
+import tempfile
 
 import pytest
 
@@ -46,19 +49,19 @@ class EnvironmentTest(unittest.TestCase):
         assert env.get_boolean('UNDEFINED') is False
 
     def test_env_vars_from_file_bom(self):
-        tmpdir = pytest.ensuretemp('env_file')
-        self.addCleanup(tmpdir.remove)
+        tmpdir = tempfile.mkdtemp('env_file')
+        self.addCleanup(shutil.rmtree, tmpdir)
         with codecs.open('{}/bom.env'.format(str(tmpdir)), 'w', encoding='utf-8') as f:
             f.write('\ufeffPARK_BOM=박봄\n')
-        assert env_vars_from_file(str(tmpdir.join('bom.env'))) == {
+        assert env_vars_from_file(str(os.path.join(tmpdir, 'bom.env'))) == {
             'PARK_BOM': '박봄'
         }
 
     def test_env_vars_from_file_whitespace(self):
-        tmpdir = pytest.ensuretemp('env_file')
-        self.addCleanup(tmpdir.remove)
+        tmpdir = tempfile.mkdtemp('env_file')
+        self.addCleanup(shutil.rmtree, tmpdir)
         with codecs.open('{}/whitespace.env'.format(str(tmpdir)), 'w', encoding='utf-8') as f:
             f.write('WHITESPACE =yes\n')
         with pytest.raises(ConfigurationError) as exc:
-            env_vars_from_file(str(tmpdir.join('whitespace.env')))
+            env_vars_from_file(str(os.path.join(tmpdir, 'whitespace.env')))
         assert 'environment variable' in exc.exconly()

